xds/internal/xdsclient/transport/transport.go

```diff
@@ -243,13 +253,15 @@ func (t *Transport) SendRequest(url string, resources []string) {
 
 func (t *Transport) newAggregatedDiscoveryServiceStream(ctx context.Context, cc *grpc.ClientConn) (adsStream, error) {
 	// The transport retries the stream with an exponential backoff whenever the
-	// stream breaks. But if the channel is broken, we don't want the backoff
-	// logic to continuously retry the stream. Setting WaitForReady() blocks the
-	// stream creation until the channel is READY.
-	//
-	// TODO(easwars): Make changes required to comply with A57:
-	// https://github.com/grpc/proposal/blob/master/A57-xds-client-failure-mode-behavior.md
-	return v3adsgrpc.NewAggregatedDiscoveryServiceClient(cc).StreamAggregatedResources(ctx, grpc.WaitForReady(true))
+	// stream breaks or if the channel is broken.
```

Comment:
Exponential backoff should not happen _whenever_ a stream breaks. It should happen _only_ when a stream breaks without ever having received a response on the stream.